helix-term/src/commands/typed.rs

```diff
@@ -1012,6 +1012,37 @@ fn sort_impl(
     Ok(())
 }
 
+fn reflow(
+    cx: &mut compositor::Context,
+    args: &[Cow<str>],
+    _event: PromptEvent,
+) -> anyhow::Result<()> {
+    // TODO: Can we instead take an Option<Cow<str>>, and then if the user
+    // doesn't pass in a selection, default to 79 characters per line?
+
+    const DEFAULT_MAX_LEN: Cow<'static, str> = Cow::Borrowed("79");
+    let max_line_len: usize = args.get(0).unwrap_or(&DEFAULT_MAX_LEN).parse()?;
+
+    let (view, doc) = current!(cx.editor);
+    let rope = doc.text();
+
+    // TODO: If only a single character is selected, we should expand the
+    // selection to whatever "object" (can we use treesitter for this? text query?)
```

Comment:
This would be tricky because of inconsistencies between grammars. For example, in Rust, each line in a comment is a separate node, so it would just end up reflowing a single line, instead of all the consecutive lines of comments together.

Reply:
Hmm, that's interesting. We can scrap this TODO, then.

Plus, if we have a way of "expanding" the selection to a comment block or something, then we basically achieve the same effect because this operates on selections.

Reply:
There's `mio` for selecting the comment block (selecting all adjacent comments nodes if the queries are specified correctly).

Reply:
`gqq` like vim to default wrapping on a single line might be a reasonable default

Reply:
The library we're using will detect a common prefix across the selected lines. This will work for most of our comments. For example, I just tested this:

```rust
// A B
// C D E F
```

and it reflowed to:

```rust
// A B C D E F
```

We fall down a bit in trying to reflow from one line to many, since the library we're using can't detect the comment prefix when it's not repeated. We also still have an issue where the library will eat blank lines and reflow across them. I'd file both of those points under "future work" on this feature.

Reply:
Hey folks! Textwrap author here :-) It's super cool to see `textwrap::refill` used. I'll be very interested in making `textwrap::unfill` (the actual engine behind `textwrap:refill`) more flexible so that it covers the necessary use-cases.

I wrote `textwrap::unfill` quickly so that it seems to do the right thing for a few languages, such as Markdown and programming languages which use `#` and `/` as line comments. But the code has not been tested much in the wild, so feedback is welcome :-)

Reply:
@mgeisler textwrap is an awesome library! I started to implement this feature myself before I was aware textwrap existed, so I have a bit of an idea of the work that went into a more comprehensive solution. And the optimal line breaking is just really cool. 

Re: `unflow`, what do you think about adding an API to provide a list of "known prefixes"? I should probably make an issue in textwrap to discuss this.

Reply:
Thanks! I think we should do exactly that: change the API of `refill` to let you specify prefixes and so on. This probably means expanding `unfill` as well  since that is the function which turns a wrapped paragraph into a single line of text.